abdiff/core/run_ab_transforms.py

```diff
@@ -155,13 +171,21 @@ def run_docker_container(
     return container
 
 
-def wait_for_containers(containers: list[Container], timeout: int = 180) -> None:
-    """Given a list of running containers, wait until all containers exit."""
-    exited_containers: list[Container] = []
+def wait_for_containers(
```

Comment:
I'm going to comment kind of high level on this function.

I have some local spike code for [TIMX-356](https://mitlibraries.atlassian.net/browse/TIMX-356) which seeks to limit how many containers can run in parallel.  This will necessarily revise a bit of the signatures of these functions, where we may choose to return a tuple like `([completed containers], [container exceptions])` from this function.

That said, I think this approach makes sense in this context.  And with this change in place, I will work to interleave this approach with any changes from TIMX-356. 

Interestingly (and encouragingly), I think the signature of this function is _more_ like what I had envisioned in TIMX-356, so feeling good about approving this as-is.

Reply:
Thank you for being flexible! This is helpful to know and I'm looking forward to reviewing the updates for TIMX-356. 🤓